src/looprestoration.rs

```diff
@@ -210,17 +227,21 @@ unsafe fn padding<BD: BitDepth>(
     let unit_w = unit_w + have_left_3 + have_right_3;
     let dst_l = &mut dst[3 - have_left_3..];
     let p = p - have_left_3;
-    let lpf = lpf.offset(-(have_left_3 as isize));
+    let lpf_off = lpf_off - (have_left_3 as isize);
     let abs_stride = stride.unsigned_abs();
 
     if have_top {
         // Copy previous loop filtered rows
+        let lpf_guard;
```

Comment:
In this case we still do. Because we're not immediately returning the borrowed value (we're first assigning it to a variable, then returning the variable), rustc is generating the temporary within the `if` block. So when I tried it without the explicit guard I got compile errors.

Reply:
Oh, okay. I noticed that sometimes happened. It's annoying.